### ISSUE TITLE
Can I build for atmosphere

### ISSUE THREAD

**Issue by sandhoners123** [DeltaLaunch/deltalaunch#3]
Does it rely on reinx or will it work on atmosphere also

**Comment by Osga21** [comment]
It checks for ReiNX on startup (see Engine.cpp)

**Comment by sandhoners123** [comment]
Also, is the source code complete. Can I remove the check and compile it

**Comment by designgears** [comment]
no, this is not complete source code, it's incomplete and crashes a lot because of unfinished code. Pretty useless to use in this state.

**Comment by sandhoners123** [comment]
Is this why I cant compile it

**Comment by sandhoners123** [comment]
error: 'struct AppletHolder' has no member named 'active'
             else if(currentApplet.active) {
                                   ^~~~~~

**Comment by designgears** [comment]
It will be released here when it's ready. https://gbatemp.net/threads/official-deltalaunch.539990/

**Comment by sandhoners123** [comment]
But how do i build it

**Comment by sandhoners123** [comment]
WIll it ever be open source

**Comment by sandhoners123** [comment]
as in not leaked

**Comment by designgears** [comment]
Eventually.

**Comment by designgears** [comment]
You can get an official test build here now. https://github.com/Reisyukaku/ReiNX/releases/tag/v2.3

**Comment by sandhoners123** [comment]
But for atmosphere?

**Comment by sandhoners123** [comment]
it has drm...

**Comment by designgears** [comment]
There is no DRM, it relies on reinx specific functions.

**Comment by sandhoners123** [comment]
In Engine.cpp, there is a check to detect reinx like this:
    //Detect reinx
    if(!Rnx::IsUsingReiNX()) {
        if(R_SUCCEEDED(romfsMountFromFsdev("/atmosphere/titles/0100000000001000>
        if(R_SUCCEEDED(romfsMountFromFsdev("/sxos/titles/0100000000001000/romfs>
        err:
        Graphics::Init(screenPos);
        SDL_Surface *surf = IMG_Load("romfs:/Graphics/BSOD.png");
        SDL_Texture *tex = Graphics::CreateTexFromSurf(surf);
        Graphics::RenderTexture(tex, screenPos);
        SDL_FreeSurface(surf);
        SDL_DestroyTexture(tex);
        Graphics::Render();
        do { Hid::KeyProcess(); } while(!Hid::Input);
        Power::Shutdown();
    }

**Comment by sandhoners123** [comment]
So one would need to patch an nsp file which I do not how to do

**Comment by designgears** [comment]
It checks ahead of time and kicks you out before you try to use functions that don't exist on other cfw. Removing that won't make it run.

**Comment by sandhoners123** [comment]
Give me one reinx only function

**Comment by designgears** [comment]
not my problem.

**Comment by sandhoners123** [comment]
What is appletRequestForeground

**Comment by sandhoners123** [comment]
is it the same as libappletRequestHomeMenu

**Comment by sandhoners123** [comment]
now i just need to link it

**Comment by sandhoners123** [comment]
And I am making a github repo for my fork of deltalaunch

**Comment by designgears** [comment]
Lol, stop, you have no clue what you're doing.

**Comment by friedkeenan** [comment]
The ReiNX only function is setting the title id for hbloader

**Comment by sandhoners123** [comment]
Then comment it out?

**Comment by sandhoners123** [comment]
or we could make a bogus rnx function

**Comment by sandhoners123** [comment]
Service in Atmosphere
